client-default/src/test/scala/com.gu.contentapi.client/GuardianContentClientTest.scala

```diff
@@ -128,4 +129,20 @@ class GuardianContentClientTest extends FlatSpec with Matchers with ScalaFutures
     }
   }
 
+  it should "paginate through all results" in {
+    var pageCount = 0
+    val query = ContentApiClient.search
+      .q("brexit")
+      .fromDate(Instant.parse("2018-05-10T00:00:00.00Z"))
+      .toDate(Instant.parse("2018-05-11T23:59:59.99Z"))
+    // http://content.guardianapis.com/search?q=brexit&from-date=2018-05-10T00:00:00.00Z&to-date=2018-05-11T23:59:59.99Z
+    // has 5 pages of results
+
+    val result = api.paginate(query) { _: SearchResponse => 
+      pageCount += 1 
+      Future.successful(())
+    }
+    
+    result.map { _ => pageCount should be (5) }
```

Comment:
Using `result.map` isn't guaranteed to succeed. It's possible that the future will fail but the test won't fail. To avoid this, we can use [`.futureValue`](http://doc.scalatest.org/3.0.1/index.html#org.scalatest.concurrent.ScalaFutures) to force the test to wait for the future and fail if it fails or times out.

ScalaTest's [asynchronous testing guide](http://www.scalatest.org/user_guide/async_testing) has more information.

Reply:
Thanks! I used `futureValue` initially but it failed because it has an implicit timeout. Surprisingly, even when extending that timeout, the test is still failing. I think it is due to the fact I am using a var. I will keep it like that and fix in next PR where I plan to replace that with a pure API.

Reply:
Gotcha. I need to re-read the documentation because at first sight I don't see where you can come to that conclusion. The very first example is similar to mine. Anyway I've now improved the API and have been able to use `futureValue`, thanks for the tip 👍